channels/generic/websocket.py

```diff
@@ -79,14 +83,16 @@ def send(self, text_data=None, bytes_data=None, close=False):
         if close:
             self.close(close)
 
-    def close(self, code=None):
+    def close(self, code=None, reason=None):
         """
         Closes the WebSocket from the server end
         """
+        message = {"type": "websocket.close"}
         if code is not None and code is not True:
-            super().send({"type": "websocket.close", "code": code})
-        else:
-            super().send({"type": "websocket.close"})
+            message["code"] = code
+        if reason:
+            message["reason"] = reason
```

Comment:
I'm not sure if those conditionals are needed, but should be fine.

Reply:
The idea is to not pollute the response with an empty "reason" which is optional in any case.